pinot-core/src/main/java/org/apache/pinot/core/data/manager/InstanceDataManager.java

```diff
@@ -78,6 +78,13 @@ void addRealtimeSegment(@Nonnull String realtimeTableName, @Nonnull String segme
   void removeSegment(@Nonnull String tableNameWithType, @Nonnull String segmentName)
       throws Exception;
 
+  /**
+   * Handles deletion of a segment from the table.
+   *
+   * This method performs book keeping of deleted segments.
+   */
+  void deleteSegment(@Nonnull String tableNameWithType, @Nonnull String segmentName);
```

Comment:
same with this method name. It is not really deleting a segment, is it?

Reply:
Will change the method name.